src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -119,6 +120,10 @@ public function normalize($object, $format = null, array $context = array())
             $data = $this->updateData($data, $attribute, $this->serializer->normalize($attributeValue, $format, $this->createChildContext($context, $attribute)));
         }
 
+        if (!\count($data)) {
+            return new EmptyObject();
```

Comment:
I fear that it's a BC break: code may rely on the old behavior and not except that this custom object is returned.
A solution may be to make this behavior optin through a context option.